symengine/derivative.cpp

```diff
@@ -503,12 +503,10 @@ class DiffImplementation
                 if (p.first != 0)
                     d[p.first - 1] = p.second * p.first;
             }
-            return make_rcp<const UnivariatePolynomial>(
-                self.get_var(), (--(d.end()))->first, std::move(d));
+            return UnivariatePolynomial::from_dict(self.get_var(), std::move(d));
```

Comment:
Yes, but since the polynomial `self` is already in canonical order, `from_dict` is not necessary